src/scripts/generate_covid_case_datamart.py

```diff
@@ -0,0 +1,131 @@
+    infection result from aggregating accross all labs.
+
+    Case Identification Process:
+    1.  Check for positive labs:
+        a.  If positive labs are found at least one case has been identified.
+        b.  The collection date of the earliest positive lab is set as the earliest
+            positive specimen collection date for the case.
+        c.  Determine case status:
+            i.  If there are any positive PCRs in the 90 day period starting on the
+                earliest positive specimen collection date then the status is confirmed.
+            ii. If there are no positive PCRs in the 90 window this implies that there
+                are only positive antigen tests and the case status is probable.
+        d. Check for positive labs more than 90 days after the initial positive test.
+            i.  If present an additional infection has been identified resulting in
+                another case. Repeat steps b-d using the collection date of the first
+                lab outside the 90 day window as the new earliest positive specimen
+                collection date.
+            ii. If not present then all covid cases for the patient have been
+                identified.
+    2.  If there are no positive labs than the patient has not been infected and no
+        cases are identified.
+
+    **** WARNING ****
+
+    As currently written this function applies an approximation of the official CDC/CTSE
+    COIVD case defintion. In its current form this function is intended as a proof of
+    concept for development purposes.
+
+    Known Limitations:
+    1.  Suspect cases resulting from antibody tests are not considered.
+    2.  No check has been implemented for a negative PCR within 48 hours of a positive
+        Antigen that would negate the case.
+    3.  No consideration has been given to how the case definition has changed over the
+        course of the pandemic.
+    """
+    case_list = []
+    patient_labs.sort_values(by=["effectiveDateTime"], ascending=True)
+    positive_labs = patient_labs.loc[patient_labs.result == "POSITIVE"].reset_index()
+    remaining_labs = len(positive_labs)
+    while remaining_labs > 0:
+        collection_date = positive_labs.effectiveDateTime[0]
+        current_infection_labs = positive_labs.loc[
+            positive_labs.effectiveDateTime.map(lambda x: (x - collection_date).days)
+            <= 90
+        ]
+        if "pcr" in current_infection_labs.type.to_list():
+            status = "confirmed"
+        else:
+            status = "probable"
+        case = patient_labs.agg(record_combination_func)
+        case["effectiveDateTime"] = collection_date
+        case["status"] = status
+        case_list.append(case)
+        positive_labs = positive_labs.loc[
+            positive_labs.effectiveDateTime.map(lambda x: (x - collection_date).days)
+            > 90
+        ].reset_index()
```

Comment:
Total non-blocker, but have we investigated where there's a more "pandas" way to do this? The functionality in here is really custom / hand-spun so there might not be, but when using pandas I know my first instinct is always to sort values and use loops when pandas can handle a lot of that for me, so I have to go back and make those changes. Just wanted to check, but if this is the best way, then totally good with me.

Reply:
🤮  okay that makes sense. Dask would almost certainly be the way to go (or potentially Vaex, that's got big data level simultaneous vectorization--I did a bunch of work for a CMS proposal using both of those), but if you've already looked into "pandas-izing" this then I don't feel the need to belabor the point here. If we make analytics more robust down the line we can revisit a more production-ready notebook anyway.

Reply:
I'm going to resolve this conversation with the understanding we will need to revisit it in the future. but seems beyond the scope of this PR.